### PR TITLE
OKTA-339612 Blazor Samples > Review Haisheng suggestions and implemen…

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
…t changes if necessary.

This Blazor sample changed to be working with one of our Resource Server samples

### CODE REVIEW

**Comment by haishengwu-okta** [comment]
Would you summarize the changes?

**Comment by andriizhegurov-okta** [comment]
@haishengwu-okta  I basically recreated the solution with only one client-side rendered project. So now:
* One project instead of three. They were Client, Server, Shared. As it's only Client now it's renamed to `okta-blazor-web-assembly-example`
* Transferred message format was changed to meet the one supported by our Resource Servers 
* Added a new `CorsRequestAuthorizationMessageHandler` class to support Cross-domain requests, because they are cross-domain now

**Review by bryanapellanes-okta — COMMENTED** [review]
This looks fine to me, but I'll defer to @laura-rodriguez for approval.

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor
Comment (laura-rodriguez) [inline web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor:26]: Nit: Fix indentation.

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor
Comment (laura-rodriguez) [inline web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor:40]: Can you remind me where you configure the client to send the access token?

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs
Comment (laura-rodriguez) [inline web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs:24]: Do we need this piece of code?

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor
Comment (andriizhegurov-okta) [reply web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Pages/FetchData.razor:40]: Client is configured in Program.cs

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs
Comment (andriizhegurov-okta) [reply web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs:24]: Currently this client is configured to work with API server on another domain, but in my opinion the most probable use case is when API server is on the same domain as frontend site and in that case commented-out code can be used.

Comment (andriizhegurov-okta) [reply web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs:24]: I think it's better to show those two cases and corresponding customization options

File: web-assembly/hosted/okta-hosted-login/README.md
Comment (laura-rodriguez) [inline web-assembly/hosted/okta-hosted-login/README.md:74]: It seems we still have instructions to run the webserver, can you please review the readme and make sure the instructions to run the example are correct? 

See this section for example:

_#### Run the web application from Visual Studio
If you run this project in Visual Studio it will start the web application on port 44314 using HTTPS. You can change this configuration in the `launchSettings.json`. Make sure that `okta-blazor-web-assembly-example.Server` is selected as your startup project._

**Review by laura-rodriguez — COMMENTED** [review]
We should rename the sample parent folder called `hosted` to whatever MS names this type of projects where you don't have a server in your solution to host the wasm project.

File: web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs
Comment (laura-rodriguez) [reply web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs:24]: I see. Maybe we can provide more details in the comment, something like "If you don't need CORS, use the following piece of code instead."

**Comment by andriizhegurov-okta** [comment]
@haishengwu-okta we need your approval here. Could you review this PR?

**Review by haishengwu-okta — APPROVED** [review]
Thank you!

**Review by laura-rodriguez — APPROVED** [review]
LGTM 👍 . Can you please also add a changelog describing all the changes? If someone has been using the current sample and pulls the latest change they will find a totally different project.

File: .travis.yaml
Comment (bretterer) [inline .travis.yaml:1]: This isn't going to work.  We need to run tests for this stuff.  Check out https://github.com/okta/samples-php/blob/develop/.travis.yml for what is being done in other languages.

**Review by bretterer — CHANGES_REQUESTED** [review]

File: .travis.yaml
Comment (laura-rodriguez) [reply .travis.yaml:1]: We're doing the same for other .NET samples. We do have e2e tests but I'm not sure when Vijet runs those tests.

**Pull request merged**
